kyber_g1.go

```diff
@@ -33,7 +33,11 @@ func NullKyberG1(dst ...byte) *KyberG1 {
 	return newKyberG1(&p, dst)
 }
 func newKyberG1(p *bls12381.PointG1, dst []byte) *KyberG1 {
-	return &KyberG1{p: p, dst: dst}
+	domain := dst
```

Comment:
I find it surprising that `dst []byte` isn't a copy here given that it's not `*[]byte`, but I tried it out in go playground and indeed it's a reference